src/extra/gfx/texture.cpp

```diff
@@ -0,0 +1,170 @@
+#include "../gfx.hpp"
+#include <gfx/texture.hpp>
+#include <gfx/error_utils.hpp>
+#include <params.hpp>
+#include <stdexcept>
+
+using namespace shards;
+namespace gfx {
+
+enum ComponentType {
+  Float,
+  Int8,
+  Int16,
+};
+
+struct TextureFormatException : public std::runtime_error {
+  TextureFormatException(ComponentType componentType, Types::TextureType_ asType)
+      : std::runtime_error(formatError(componentType, asType)) {}
+
+  static std::string formatError(ComponentType componentType, Types::TextureType_ asType) {
+    return fmt::format("Image with component type '{}' can not be converted to texture type '{}'",
+                       magic_enum::enum_name(componentType), magic_enum::enum_name(asType));
+  }
+};
+
+struct TextureShard {
+  static inline shards::Types InputTypes{{CoreInfo::ImageType}};
+  static SHTypesInfo inputTypes() { return InputTypes; }
+  static SHTypesInfo outputTypes() { return Types::Texture; }
+  static SHOptionalString help() { return SHCCSTR("Creates a texture from an image"); }
+
+  TexturePtr texture;
+  OwnedVar textureVar;
+
+  PARAM_VAR(asType_, "Type", "Type to interpret image data as. (Default: UNormSRGB for RGBA8 images, UNorm for other formats)",
+            {Types::TextureType, CoreInfo::NoneType});
```

Comment:
Do we need none here? Just a texturetype default should be ok no?

Reply:
That's the default in that case, right?
In previous PRs it was suggested that I should add NoneType if it can be left empty.

Reply:
I think with enums we don't need None. it's an enum right?
Default will be the first element of enum or 0 (actually you decide C++ side) so far for enums that's what I did
UX wise is better too

Reply:
Fixed